control/control.go

```diff
@@ -933,9 +1061,162 @@ func (p *pluginControl) UnsubscribeDeps(taskID string, mts []core.Metric, plugin
 		}
 	}
 
+	td.results = mts
+	td.plugins = plugins
+
+	p.taskView[taskID] = td
+
+	// just debug logging
```

Comment:
Do we want to loop on debug logging even when debug is not set.  Seems like we will eat the cost of these loops even if nothing will happen in them.